ql/termstructures/credit/probabilitytraits.hpp

```diff
@@ -264,6 +267,96 @@ namespace QuantLib {
         static Size maxIterations() { return 30; }
     };
 
+
+
+
+    /*! Piecewise (deterministic) plus affine (stochastic) terms composed
+     hazard rate
+     */
+    struct AffineHazardRate {
```

Comment:
I would move this class to a file in `ql/experimental/credit` to avoid having the core library depend on experimental files.